core/src/main/java/org/apache/iceberg/view/BaseViewHistoryEntry.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import java.util.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+public class BaseViewHistoryEntry implements ViewHistoryEntry {
+  private final long timestampMillis;
+  private final int versionId;
+
+  static ViewHistoryEntry of(long timestampMillis, int versionId) {
```

Comment:
why do we need a static method if it is just delegating to the constructor?

Reply:
It's not needed, just a part of the original CR. We can just make the constructor public and use that

Reply:
I see. I can see the constructor being potentially protected, we can keep it like that for now, but the static method seems unnecessary